### PR TITLE
HDDS-5403 Support list node based on NodeOperationalState and NodeState options in datanode list CLI

### PR DESCRIPTION
**Pull request by cchenax**
## What changes were proposed in this pull request?

Currently, printTopology CLI display all DN information as output.
This task is to add NodeOperationalState and NodeState as the CLI options to the command, and display the DN accordingly.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-5403

## How was this patch tested?

bin/ozone admin datanode list --NodeOperationalState IN_SERVICE
bin/ozone admin datanode list --NodeState HEALTHY

### CODE REVIEW

**Review by ayushtkn — COMMENTED** [review]
Can you extend a unit test covering your changes

File: hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/datanode/ListInfoSubcommand.java [hunk 1]
```diff
@@ -51,6 +51,16 @@
       defaultValue = "")
   private String uuid;
 
+  @CommandLine.Option(names = {"--NodeOperationalState"},
+      description = "Show info by datanode NodeOperationalState.",
```
Comment (ChenSammi) [inline hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/datanode/ListInfoSubcommand.java:55]: Please add all available NodeOperationalState values in the description to asssit user's better choice.

File: hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/datanode/ListInfoSubcommand.java [hunk 2]
```diff
@@ -51,6 +51,16 @@
...
+      defaultValue = "")
+  private String nodeOperationalState;
+
+  @CommandLine.Option(names = {"--NodeState"},
+      description = "Show info by datanode NodeState.",
```
Comment (ChenSammi) [inline hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/datanode/ListInfoSubcommand.java:60]: Same as above.

File: hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/datanode/ListInfoSubcommand.java [hunk 3]
```diff
@@ -51,6 +51,19 @@
       defaultValue = "")
   private String uuid;
 
+  @CommandLine.Option(names = {"--NodeOperationalState"},
+      description = "Show info by datanode NodeOperationalState." +
+          "IN_SERVICE DECOMMISSIONING DECOMMISSIONED ENTERING_MAINTENANCE" +
```
Comment (ChenSammi) [inline hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/datanode/ListInfoSubcommand.java:56]: Please change the description to the following format "Show info by datanode NodeOperationalState(IN_SERVICE, DECOMMISSIONING, DECOMMISSIONED, ENTERING_MAINTENANCE, IN_MAINTENANCE)."

File: hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/datanode/ListInfoSubcommand.java [hunk 4]
```diff
@@ -51,6 +51,19 @@
...
+          "IN_MAINTENANCE",
+      defaultValue = "")
+  private String nodeOperationalState;
+
+  @CommandLine.Option(names = {"--NodeState"},
+      description = "Show info by datanode NodeState." +
+      " HEALTHY STALE DEAD",
```
Comment (ChenSammi) [inline hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/datanode/ListInfoSubcommand.java:63]: Same as above.

File: hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/datanode/ListInfoSubcommand.java (hunk 3 above)
Comment (cchenax) [reply hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/datanode/ListInfoSubcommand.java:56]: ok

File: hadoop-ozone/dist/src/main/smoketest/admincli/datanode.robot [hunk 5]
```diff
@@ -34,6 +34,14 @@ Filter list by UUID
     ${count} =          Get Length   ${lines}
     Should Be Equal As Integers    ${count}    1
 
+Filter list by NodeOperationalState
+    ${output} =         Execute      ozone admin datanode list --NodeOperationalState IN_SERVICE
+    Should contain      ${output}    Datanode:
```
Comment (ChenSammi) [inline hadoop-ozone/dist/src/main/smoketest/admincli/datanode.robot:39]: Can you add a datanode UUID to the "Datanode:" string ?

File: hadoop-ozone/dist/src/main/smoketest/admincli/datanode.robot [hunk 6]
```diff
@@ -34,6 +34,14 @@ Filter list by UUID
...
+
+Filter list by NodeState
+    ${output} =         Execute      ozone admin datanode list --NodeState HEALTHY
+    Should contain      ${output}    Datanode:
```
Comment (ChenSammi) [inline hadoop-ozone/dist/src/main/smoketest/admincli/datanode.robot:43]: Same as above

**Comment by ChenSammi** [comment]
The last patch LGTM，+1.

**Pull request merged**